core/src/test/scala/org/typelevel/paiges/PaigesTest.scala

```diff
@@ -46,6 +46,20 @@ object PaigesTest {
       case Nest(_, d) => twoRightAssociated(d)
     }
   }
+
+  // Definition of `fill` from the paper
+  def fillSpec(ds: List[Doc]): Doc = {
+    import Doc._
+    ds match {
+      case Nil => empty
+      case x :: Nil => x
```

Comment:
I thought we suggested that `x.grouped` was a better definition here?

I can't find it now, but someone made a Idris or Agda implementation of that paper and found one of the claims to be false when they were trying to prove in the types all the claims. I wonder if it might have been related to fill, since again we needed to put `x.grouped` to pass a law we added earlier.

Reply:
Yes, we did say that.  I just wanted a minimal CR.  If you'd like, I'm happy to change both the implementation and the spec to `_.grouped`.

Reply:
didn't we change the implementation already?

Reply:
No.

Reply:
I just updated both here.  

At one (a few?) points I had it in one of the many revisions CRs for this `FlatAlt` change.